packages/driver/src/cypress/log.ts

```diff
@@ -13,11 +13,32 @@ import $errUtils from './error_utils'
 const groupsOrTableRe = /^(groups|table)$/
 const parentOrChildRe = /parent|child|system/
 const SNAPSHOT_PROPS = 'id snapshots $el url coords highlightAttr scrollBy viewportWidth viewportHeight'.split(' ')
-const DISPLAY_PROPS = 'id alias aliasType callCount displayName end err event functionName hookId instrument isStubbed group message method name numElements showError numResponses referencesAlias renderProps state testId timeout type url visible wallClockStartedAt testCurrentRetry'.split(' ')
+const DISPLAY_PROPS = 'id alias aliasType callCount displayName end err event functionName groupLevel hookId instrument isStubbed group message method name numElements showError numResponses referencesAlias renderProps state testId timeout type url visible wallClockStartedAt testCurrentRetry'.split(' ')
 const BLACKLIST_PROPS = 'snapshots'.split(' ')
 
 let counter = 0
 
+export interface LogConfig {
+  // defaults to command
+  instrument?: 'agent' | 'command' | 'route'
+  // name of the log
+  name?: string
+  // additional information to include in the log if not overridden
+  // the render props message
+  // defaults to command arguments for command instrument
+  message?: string
+  // the JQuery element for the command. This will highlight the command
+  // in the main window when debugging
+  $el?: JQuery
+  // whether or not to show the log in the Reporter UI or only
+  // store the log details on the command and log manager
+  emitOnly?: boolean
+  // whether or not to start a new log group
+  groupStart?: boolean
+  // timeout of the group command - defaults to defaultCommandTimeout
+  timeout?: number
+}
```

Comment:
I think this should be combined somehow with the `LogConfig` defined in `internal-types.d.ts`:
https://github.com/cypress-io/cypress/blob/938f72ac3b2b03eb8112a52e28b32596d35d7ceb/packages/driver/types/internal-types.d.ts#L51-L70